### PR TITLE
add public helper `ray_intersection_over_mesh` to use directly ray casting without going through the ECS

### PR DESCRIPTION
**Pull request by mockersf**
Having direct access to this method is useful in some cases where one need to check a varying number of rays depending on the situation.

Also added an example using it

### CODE REVIEW

File: examples/ray_intersection_over_mesh.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .insert(PathStatus);
+}
+
+// Marker struct for the text
+struct PathStatus;
+// Marker struct for the ground, used to et cursor position
```
Comment (aevyrie) [inline examples/ray_intersection_over_mesh.rs:81]: "et" - "get" ?

File: examples/ray_intersection_over_mesh.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+struct Ground;
+// Marker struct for the path origin, showed by a yellow sphere
```
Comment (aevyrie) [inline examples/ray_intersection_over_mesh.rs:83]: Really minor grammar nit, should be "shown"

**Review by aevyrie — COMMENTED** [review]
Found some _very_ minor grammar nits in comments. 🙂 

Otherwise, this looks good. This has been something I've wanted to do for a while - really appreciate that you took it on! The example is pretty complex, but I like how it's a practical example of how it might be used.

**Comment by mockersf** [comment]
I wanted to have an example where it would somewhat make sense to use it instead of relying on the system update, it ended up more complex that I would have hoped...

**Pull request merged**